command/rootchain/deploy/deploy.go

```diff
@@ -560,15 +560,13 @@ func deployContracts(outputter command.OutputFormatter, client *jsonrpc.Client,
 					return fmt.Errorf("deployment of %s contract failed", contract.name)
 				}
 
-				resultsLock.Lock()
-				defer resultsLock.Unlock()
-
+				var deployResults []*deployContractResult
```

Comment:
Nitpick:
the worst-case scenario is to have 2 items in the `deployResults` slice, so I'd suggest specifying a capacity

```suggestion
				deployResults := make([]*deployContractResult, 0, 2)
```